framework/src/multiapps/MultiApp.C

```diff
@@ -1108,8 +1108,31 @@ MultiApp::createApp(unsigned int i, Real start_time)
   const auto & input_file = _input_files[input_index];
 
   // create new parser tree for the application and parse
-  app_params.set<std::shared_ptr<Parser>>("_parser") = std::make_shared<Parser>(input_file);
+  auto parser = std::make_unique<Parser>(input_file);
 
+  if (input_file.size())
+  {
+    parser->parse();
+    auto app_type = parser->getAppType();
+    if (app_type.empty() && _app_type.empty())
```

Comment:
Yep, app type is not provided in application block or multiApp block, but this rarely happens I would say because for Application and multiApp there are default app type set up. The code has to be very wrong to hit this error. Maybe I should delete it.